LWCOSC/force-app/main/default/classes/ProductController.cls

```diff
@@ -0,0 +1,35 @@
+public with sharing class ProductController {
+    
+    @AuraEnabled(cacheable=true)
+    public static List<Product2> getProducts(String searchKey, String sortBy, String sortDirection) {
+        ProductServiceImpl proSerImpl = new ProductServiceImpl();
+        List<Product2> prodList = new List<Product2>();
+        try{
+            prodList= proSerImpl.getProducts(searchKey, sortBy, sortDirection);
+            return prodList;
+        }
+        catch(Exception ex){
+            System.System.debug('Exception in ProductController at method getProducts = '+ex.getMessage());
+            return prodList;
```

Comment:
Should we return any thing if there was any exception?

Reply:
Instead you should re-throw the exception wrapped in "AuraHandledException" class

Reply:
Done